### PR TITLE
IA-2699 Allow to limit authentication to third party authentication

### PR DESCRIPTION
**Pull request by madewulf**
Adding an environment variable DISABLE_PASSWORD_LOGINS that allows to disable logins using password (to only allow SSO )

Related JIRA tickets : IA-2699 (and sub stories)

I also removed a LOT of print() in the codebase so that the test outputs are clean. 

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [x] My migrations file are included
- [x] Are there enough tests
- [x] Documentation has been included (for new feature)

## Doc
- under /docs

## How to test
Activate the environment variable. 

Go to /login  or /admin and check that no password field is available to fill
 /api/token/ should give you a 404 too

## Print screen / video

CleanShot 2024-03-15 at 18 11 52@2x

### CODE REVIEW

File: hat/urls.py [hunk 1]
```diff
@@ -10,24 +10,37 @@
...
+
+urlpatterns = urlpatterns + [
     path("", RedirectView.as_view(pattern_name="dashboard:home_iaso", permanent=False), name="index"),
     path("_health/", health),
     path("_health", health),  # same without slash otherwise AWS complain about redirect
     path("health/", health),  # alias since current apache config hide _health/
     path("accounts/", include("django.contrib.auth.urls")),
     path("accounts/", include("allauth.urls")),
+    path("login/", auth.views.LoginView.as_view(template_name="iaso/login.html"), name="login"),
+    path("admin/login/", auth.views.LoginView.as_view(template_name="iaso/login.html"), name="login"),
```
Comment (kemar) [inline hat/urls.py]:
```suggestion
```

It seems this should be removed as it's already defined in the above condition.

File: iaso/tests/api/test_token.py [hunk 2]
```diff
@@ -6,6 +6,38 @@
 from iaso.test import APITestCase
 
 
+class DisableLoginTokenAPITestCase(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        data_source = m.DataSource.objects.create(name="counsil")
+        version = m.SourceVersion.objects.create(data_source=data_source, number=1)
+        star_wars = m.Account.objects.create(name="Star Wars", default_version=version)
+        cls.yoda = cls.create_user_with_profile(username="yoda", account=star_wars)
```
Comment (kemar) [inline iaso/tests/api/test_token.py]: You told me we should stop using names such as `star_wars` etc. ;)

File: iaso/tests/api/test_token.py [hunk 3]
```diff
@@ -6,6 +6,38 @@
...
+
+    def test_acquire_token_and_authenticate_when_passwords_disabled(self):
+        """Test that token authentication is not possible when passwords are disabled"""
+
+        def reload_urlconf():
+            import importlib
+            from django.urls import clear_url_caches
+
+            clear_url_caches()
+            importlib.reload(importlib.import_module("hat.urls"))
```
Comment (kemar) [inline iaso/tests/api/test_token.py]: I thought reloading `hat.urls` would be enough, but it seems not.

File: iaso/tests/api/test_token.py [hunk 4]
```diff
@@ -6,6 +6,38 @@
...
+            importlib.reload(importlib.import_module("iaso.urls"))
+
+        with self.settings(DISABLE_PASSWORD_LOGINS=True):
+            reload_urlconf()
+            response = self.client.post(
+                f"/api/token/", data={"username": "yoda", "password": "IMomLove"}, format="json"
+            )
+            self.assertEqual(response.status_code, 404)
+
+        reload_urlconf()
```
Comment (kemar) [inline iaso/tests/api/test_token.py]: This is useful and could be more generic.

I tried but It's harder than it looks.

I suggest this https://github.com/BLSQ/iaso/pull/1194

**Review by kemar — APPROVED** [review]
LGTM
I suggest one little improvement in https://github.com/BLSQ/iaso/pull/1194

File: iaso/tests/api/test_token.py (hunk 2 above)
Comment (madewulf) [reply iaso/tests/api/test_token.py]: Correct! I copy pasted these, I shouldn't have. Will correct

File: iaso/tests/api/test_token.py (hunk 3 above)
Comment (madewulf) [reply iaso/tests/api/test_token.py]: No, that was the problem we resolved with @mathvdh on Friday. Having the url cache reset after the test is also a challenge.

**Pull request merged**